db/migrate/20150625220141_fix_serialized_reports_for_rails_four.rb

```diff
@@ -126,34 +121,74 @@ def parts
 
   def up
     say_with_time("Converting MiqReportResult#report to a serialized hash") do
-      MiqReportResult.where('report IS NOT NULL').find_each do |rr|
-        val = rr.serialize_report_to_hash(rr.read_attribute(:report), self)
-        rr.update_attribute(:report, val) if val
+      base_relation = MiqReportResult.where('report IS NOT NULL')
+
+      say_batch_started(base_relation.size)
+
+      base_relation.find_in_batches do |batch|
+        MiqReportResult.transaction do
+          batch.each do |rr|
+            val = Serializer.serialize_report_to_hash(rr.report, rr.class, rr.id, self)
+            rr.update_attribute(:report, val) if val
+          end
+        end
+        say_batch_processed(batch.size)
       end
     end
 
     say_with_time("Converting BinaryBlob report results to a serialized hash") do
-      BinaryBlob.includes(:resource).where(:resource_type => 'MiqReportResult').find_each do |bb|
-        if bb.resource
-          val = bb.serialize_report_to_hash(bb.binary, self)
-          bb.binary = val if val
+      # Pre-autoload classes that appear in reports
+      MiqExpression
+      Ruport
+
+      all_ids = BinaryBlob
+                  .where(:resource_type => 'MiqReportResult')
+                  .select(:id, :resource_id, :resource_type)
+                  .includes(:resource)
```

Comment:
not sure if it makes a difference in the grander scheme...

Since you only want to verify that the `resource_id` exists in the id, why bring back the resource records?

```diff
-.select(:id, :resource_id, :resource_type)
+.select(:id)
- .includes(:resource)
- .select(&:resource)
+ .where("exists (select id from miq_report_results where miq_report_results.id = binary_blobs.resource_id)")
```

In the grand scheme of th